core/nginx/conf/nginx.conf

```diff
@@ -34,6 +34,25 @@ http {
       ''      $scheme;
     }
 
+    {% if KUBERNETES_INGRESS != 'true' and TLS_FLAVOR in [ 'letsencrypt', 'cert' ] and not TLS_ERROR %}
```

Comment:
Why is `TLS_ERROR` excluded?

If I understand this change correctly, there is no way to recover from `TLS_ERROR`, when `TLS_FLAVOR=letsencrypt`.
This is due to the only other place with a acme-challenge proxy being reserved for `TLS_FLAVOR=mail-letsencrypt` in L102.

I suspect the `TLS_ERROR` checks in this line and in L71 could be deleted.

This would prevent http access while `TLS_ERROR=yes`, because of the `return 301`, but I'd honestly consider it a feature to prevent unencrypted access when TLS is actually intended.